### ISSUE TITLE
Source kind fallback during inspect and remove

### ISSUE THREAD

**Issue by phillebaba** [zarf-dev/zarf#2894]
### Describe what should be investigated or refactored

The logic for getting the package source differs slightly depending on the function that is being called. For the most part we will try to determine the package source kind based on the file name. But in the case of inspecting and removing packages we allow falling back to a special cluster source when the source kind could not be determined.

https://github.com/zarf-dev/zarf/blob/8100c698d92dd1e3fed450d548b2fea1d08cf32e/src/cmd/package.go#L314-L325

The function name lies slightly as the fallback is to create a cluster source and returning nil means that the source will be created at a later time when creating the packager.

https://github.com/zarf-dev/zarf/blob/8100c698d92dd1e3fed450d548b2fea1d08cf32e/src/pkg/packager/common.go#L100-L105

This type of logic makes it hard to follow the special cases as the relationship between the fallback function and the purpose it serves is not easy to initially determine. The functionality however is critical and could be easy to miss is someone were to refactor the code.

On top of this the delete logic is a bit ambiguous due to how it is implemented. When the cluster source is used as a fallback the name of the package source will be used to lookup the package and return it. If a tar file is passed as the package source the metadata will be fetched and the package name is used.  This package name is then used to get the deployed package in the cluster. In theory this means that running delete on a different package artifact with the same name and same name of components could result in the deletion of an unrelated package.

https://github.com/zarf-dev/zarf/blob/8100c698d92dd1e3fed450d548b2fea1d08cf32e/src/pkg/packager/remove.go#L74-L84

**Comment by brandtkeller** [comment]
We'll want to be cognizant of changes in #2407 regarding package removal and workflow as it overlaps with this issue.

**Comment by AustinAbro321** [comment]
We removed this logic during the packager refactor
